submissions/maxim-zabolotny/dom-practise/app.js

```diff
@@ -0,0 +1,128 @@
+        },
+        {
+            id: 4,
+            name: "Starship",
+            img: 'https://techcrunch.com/wp-content/uploads/2019/09/Starship-Mk1-Day.jpg',
+            description: `Super-heavy reusable launch vehicle developed by the private American company SpaceX since 2012. Starship is designed to launch more than 100 tons of payload into low orbit. As part of the Starship project, SpaceX is also developing a Super Heavy launch vehicle to launch a spacecraft into low-earth orbit from the ground and a Starship-based "tanker" for refueling in orbit for deep space missions. According to SpaceX CEO and Chief Engineer Elon Musk, Starship will in the future replace Falcon 9 and Falcon Heavy, the cargo and manned Dragon V2.
+            The goal of SpaceX is to launch Starship: to Mars with cargo in 2022, followed by a manned flight in 2024.`
+        },
+    ],
+    activeIndex: 0,
+}
+
+const menuContainer = document.querySelector('.menu-container');
+const menuUl = document.querySelector('.menu');
+const toggleMenuButton = document.querySelector('.btn-toogle-menu');
+const wrapper = document.querySelector('.wrapper');
+const title = document.querySelector('.title');
+const image = document.querySelector('.picture');
+const description = document.querySelector('.description');
+const locationLine = document.querySelector('.habitat-line');
+const locationArea = document.querySelector('.habitat');
+const attractionMethod = document.querySelector('.method');
+
+const getViewWidth = () => Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);
+const maxAutoHideMenuWidth = 820;
+
+const toggleMenu = (e) => {
+    e.target.classList.toggle("closed");
+    menuContainer.classList.toggle("opened") ;
+    wrapper.classList.toggle("wide");
+}
+
+const menuItemClickListener = (event)=>{
+
+    if (event.target.matches('.menu-item')) {
+      state.activeIndex = Number(event.target.parentNode.id);
+      renderPage(state);
+  
+      if (getViewWidth() < maxAutoHideMenuWidth) {
+        document.querySelector('.btn-toogle-menu').click(); 
+      }
+    };
+  };
+  
+  const renderMenu = (state)=>{
+    
+    menuUl.innerHTML = "";
+  
+    state.rockets.forEach((item,index)=>{
+      const li = document.createElement("li");   
+      li.id = index;
+  
+      const a = document.createElement("a");
+      a.innerHTML = item.name;
+      a.classList.add("menu-item");
+      a.href = "javascript:void(0)";
+      
+      if (index===state.activeIndex) {
+        a.classList.toggle("active");
+      };
+  
+      li.appendChild(a);
+      menuUl.appendChild(li);
+    });
+  };
+  
+  const showTextContent = ()=>{
+    setTimeout(()=> {
+      description.classList.remove("transparent");
+    }, 150);
```

Comment:
Why you need this delay?

Reply:
this function waits until the image is loaded 
and after that with a slight delay, it displays the description for it